purchase_procurement_analytic/tests/test_purchase_procurement_analytic.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# © 2016 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp.tests.common import TransactionCase
+
+
+class TestPurchaseProcurementAnalytic(TransactionCase):
+    """ Use case : Prepare some data for current test case """
+    def setUp(self):
+        super(TestPurchaseProcurementAnalytic, self).setUp()
+
+        self.product = self.env.ref('product.product_product_8')
+        self.analytic_account = self.env['account.analytic.account'].create({
+            'name': 'Test Analytic Account',
+            'type': 'contract',
+        })
+        self.procurement = self.env['procurement.order'].create({
+            'name': 'Procurement test',
+            'product_id': self.product.id,
+            'product_uom': self.product.uom_id.id,
+            'warehouse_id': self.env.ref('stock.warehouse0').id,
+            'location_id': self.env.ref('stock.stock_location_stock').id,
+            'route_ids': [
+                (6, 0, [self.env.ref('purchase.route_warehouse0_buy').id])],
+            'product_qty': 1.0,
+            'account_analytic_id': self.analytic_account.id,
+        })
+
+    def test_procurement_to_purchase(self):
+        # Run procurement
+        self.procurement.run()
+
+        # Search purchase order line generate by procurement run
+        pol = self.env['purchase.order.line'].search(
+            [('account_analytic_id', '=', self.analytic_account.id)])
+        self.assertTrue(pol)
```

Comment:
Test here that the purchase orders lines are not mixed if they have different analytic accounts (or one is empty). 

The other flow for method `_prepare_procurement_from_move` is also interested to be tested.